src/devices/Sgp30/Sgp30.cs

```diff
@@ -0,0 +1,279 @@
+        protected I2cDevice _i2cDevice;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Sgp30"/> class.
+        /// </summary>
+        public Sgp30(I2cDevice i2cDevice)
+        {
+            _i2cDevice = i2cDevice;
+        }
+
+        /// <summary>
+        /// Gets the Serial ID of the SGP30 sensor.
+        /// </summary>
+        /// <returns>Device Serial ID.</returns>
+        /// <exception cref="ChecksumFailedException">Thrown if checksum validation fails.</exception>
+        public ushort[] GetSerialId()
+        {
+            Span<byte> resultBuffer = stackalloc byte[10];
+            _i2cDevice.Write(new ReadOnlySpan<byte>(new byte[] { (byte)((SGP30_GET_SERIAL_ID & 0xFF00) >> 8), (byte)(SGP30_GET_SERIAL_ID & 0x00FF) }));
+            _i2cDevice.Read(resultBuffer.Slice(1));
+            byte[] resultArray = resultBuffer.ToArray();
+
+            ushort[] serialValues = new ushort[]
+            {
+                (ushort)(resultArray[1] << 8 | resultArray[2]),
+                (ushort)(resultArray[4] << 8 | resultArray[5]),
+                (ushort)(resultArray[7] << 8 | resultArray[8])
+            };
+
+            byte[] checksums = new byte[]
+            {
+                CalculateChecksum(serialValues[0]),
+                CalculateChecksum(serialValues[1]),
+                CalculateChecksum(serialValues[2]),
+            };
+
+            if (checksums[0] != resultArray[3] || checksums[1] != resultArray[6] || checksums[2] != resultArray[9])
+            {
+                throw new ChecksumFailedException();
+            }
+
+            return serialValues;
+        }
+
+        /// <summary>
+        /// Get the device featureset version.
+        /// </summary>
+        /// <returns>Device Features.</returns>
+        public ushort GetFeaturesetVersion()
+        {
+            Span<byte> resultBuffer = stackalloc byte[4];
+            _i2cDevice.Write(new ReadOnlySpan<byte>(new byte[] { (byte)((SGP30_GET_FEATURESET_VERSION & 0xFF00) >> 8), (byte)(SGP30_GET_FEATURESET_VERSION & 0x00FF) }));
+            _i2cDevice.Read(resultBuffer.Slice(1));
+            byte[] resultArray = resultBuffer.ToArray();
+            ushort result = (ushort)(resultArray[1] << 8 | resultArray[2]);
+            byte checksum = CalculateChecksum(result);
+
+            if (checksum != resultArray[3])
+            {
+                throw new ChecksumFailedException();
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Initialises the SGP30 for measurement. NOTE: This call will block for up to 20 seconds.
+        /// </summary>
+        /// <returns>Initial SGP30 measurement result.</returns>
+        public Sgp30Measurement InitialiseMeasurement()
+        {
+            _i2cDevice.Write(new ReadOnlySpan<byte>(new byte[] { (byte)((SGP30_INITIALISE_AIR_QUALITY & 0xFF00) >> 8), (byte)(SGP30_INITIALISE_AIR_QUALITY & 0x00FF) }));
+            Thread.Sleep(10);
```

Comment:
Can you point in the documentation where you have to wait for 10 milliseconds? this helps later on on the debug or changes that may have to be done on this code